### ISSUE TITLE
Reproducing Moving Mnist with ConvLSTM

### ISSUE THREAD

**Issue by black0017** [ndrplz/ConvLSTM_pytorch#19]
Hello I am trying to reproduce the results of the paper in moving Mnist.
I developed my own implementation but it didnt converge. Now I will try with your proposed implementation.

I was wondering have you managed to use it in any dataset link Moving Mnist?

Does it converge?

By the way, I am not 100% sure how you implement the input to state transitions and final state to output transition. Do you do it via 2D(in the timestep) or 3D convs (in all the sequence)

Any ideas?
N.A.

**Comment by DavideA** [comment]
Hi,

If I remember correctly, we ran a training on moving MNIST as a safety check. 
We managed to forecast future frames successfully but we didn't measure any metric.

Also note that the dynamics implemented in this repo are slightly different from the ones in the original paper (please check the readme).

D

**Comment by black0017** [comment]
I am aware of the differences in implementation. Thanks for the quick answer and notifying me about that. I also want a proof of concept(safety check), as you say to predict a couple of future frames. 

So, could you provide any info on how did you did it ? I mean did you add any layers before you Conv LSTM class or after?

Any help would be highly appreciated as I am kind of stuck. 

The official paper does not help so much.

Thanks again!

**Comment by tcapelle** [comment]
If you want a working example check my repo: https://github.com/tcapelle/moving_mnist

**Comment by black0017** [comment]
@tcapelle  cool I was thinking to make a pytorch notebook also on this. thanks
